src/btree/bt_random.c

```diff
@@ -169,27 +169,22 @@ __wt_row_random_leaf(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
  *	Find a random page in a tree for either sampling or eviction.
  */
 int
-__wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, bool eviction)
+__wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags)
 {
 	WT_BTREE *btree;
 	WT_DECL_RET;
 	WT_PAGE *page;
 	WT_PAGE_INDEX *pindex;
 	WT_REF *current, *descent;
-	uint32_t flags, i, entries, retry;
+	uint32_t i, entries, retry;
+	bool eviction;
 
 	*refp = NULL;
 
 	btree = S2BT(session);
 	current = NULL;
 	retry = 100;
-
-	/* Eviction should not be tapped to do eviction. */
-	if (eviction)
-		flags = WT_READ_CACHE | WT_READ_NO_EVICT | WT_READ_NO_GEN |
-		    WT_READ_NO_WAIT | WT_READ_NOTFOUND_OK | WT_READ_RESTART_OK;
-	else
-		flags = WT_READ_RESTART_OK;
+	eviction = LF_ISSET(WT_READ_CACHE);
```

Comment:
A comment here about why `WT_READ_CACHE` means eviction would be helpful.

Reply:
@michaelcahill can you provide some assistance with this?

Reply:
I'd add a comment saying something like "This function is called by eviction to find a random page in cache.  That case is indicated by the WT_READ_CACHE flag.  Ordinary lookups in a tree will read pages into cache as needed."